pennylane/pauli/utils.py

```diff
@@ -440,11 +440,32 @@ def pauli_word_to_string(pauli_word, wire_map=None):
         raise TypeError(f"Expected Pauli word observables, instead got {pauli_word}")
     if isinstance(pauli_word, qml.Hamiltonian):
         # hamiltonian contains only one term
-        pauli_word = pauli_word.ops[0]
-    elif isinstance(pauli_word, SProd):
-        pauli_word = pauli_word.base
-    if isinstance(pauli_word, Prod):
-        pauli_word = Tensor(*pauli_word.operands)
+        return _pauli_word_to_string_legacy(pauli_word, wire_map)
```

Comment:
Updating pauli_word_to_string function saparating and unnecessarily complicated old version and the new (needed because Hamiltonian doesnt have a pauli_rep at the moment)

Reply:
Yep, the line `pauli_word = Tensor(*pauli_word.operands)` causes problem with ` tests/ops/op_math/test_exp.py`. Hopefully, this should fix it!